projects/components/src/navigation/navigation.config.ts

```diff
@@ -10,6 +10,7 @@ export interface NavItemLinkConfig {
   iconSize?: IconSize;
   label: string;
   matchPaths: string[]; // For now, default path is index 0
+  queryParams?: QueryParamObject;
```

Comment:
You probably want to replace match paths with a more advanced descriptor. This would add support for sending a query param on click, but matching would be broken. There was an existing issue that @arjunlalb might recall too about multi-segment path matches.

Reply:
Yep. We've had matching issues for nav items. We only use first item for matching/nav item highlighting.